lib/spack/spack/config.py

```diff
@@ -808,8 +790,35 @@ def _config():
     cfg = Configuration()
 
     # first do the builtin, hardcoded defaults
-    defaults = InternalConfigScope('_builtin', config_defaults)
-    cfg.push_scope(defaults)
+    builtin = InternalConfigScope('_builtin', config_defaults)
+    cfg.push_scope(builtin)
+
+    # Builtin paths to configuration files in Spack
+    configuration_paths = [
+        # Default configuration scope is the lowest-level scope. These are
+        # versioned with Spack and can be overridden by systems, sites or users
+        configuration_defaults_path,
+    ]
+
+    # System configuration is per machine.
+    # This is disabled if user asks for no local configuration.
+    if not os.getenv("SPACK_DISABLE_LOCAL_CONFIG"):
```

Comment:
```suggestion
    if "SPACK_DISABLE_LOCAL_CONFIG" not in os.environ:
```

Forcing the user to provide a value suggests that the user should pass a truthy value and then by extension falsy values `SPACK_DISABLE_LOCAL_CONFIG=0` may turn the feature off, which is not the case. So, maybe check for existence instead? This is different from SPACK_*_CONFIG_PATH since we care about the value there, and an empty path does not make sense.

Reply:
Would you rather support 0, false, and empty as possible false values? Seems cleaner to me.

Reply:
Hm, I think set/unset is quite a clear boolean, whereas it's unclear what strings are false-y, so my preference is to just stick to set/unset :thinking: we currently have SPACK_COLOR=always/never/auto which mirrors command line arg values, and SPACK_STACKTRACE which also maps to a command line argument and uses set/unset, not truthy/falsy strings